src/scriptMode.ts

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env node
+
+// add script mode flag in not present
+if (!(process.argv.includes('-s') || process.argv.includes('--script-mode'))) {
+  process.argv.splice(2, 0, '-s')
```

Comment:
I’m not super comfortable with this. I think we could probably export a function from `bin` instead, and use it here, to avoid mutating `process.argv`. But LGTM anyway, I’ll make any changes after we merge.

Reply:
Alright, I just wanted to achieve it with minimal changes, but there's certainly a more proper way to do it. The bin can be exported as function and have a code that autoruns itself if it is being called as main module:
```js
function bin(args) {
    // code
}

module.exports = bin;

if (require.main === module) {
    bin();
}
```